src/status_im2/common/confirmation_drawer/view.cljs

```diff
@@ -33,10 +33,14 @@
       (let [{:keys [group-chat chat-id public-key color
                     profile-picture name]} context
             id                             (or chat-id public-key)
-            display-name                   (or
-                                            (when-not group-chat
-                                              (rf/sub [:contacts/contact-name-by-identity id]))
-                                            name)
+            contact-name-by-identity       (when-not group-chat
+                                             (rf/sub [:contacts/contact-name-by-identity id]))
+            display-name                   (cond
+                                             (= contact-name-by-identity
+                                                "Unknown") name
+                                             (= contact-name-by-identity
+                                                nil)       name
+                                             :else         contact-name-by-identity)
```

Comment:
We probably should prefer passed name over subscription. Should we try something like
```
(if (and name (not (starts-with 0x)) 
    name
    (rf/sub [:contacts/contact-name-by-identity id]))
```